test/unit/three_scale/analytics/user_tracking_test.rb

```diff
@@ -8,6 +8,32 @@ def setup
     @user = FactoryBot.build_stubbed(:user, account: account, email: 'foo@example.net')
   end
 
+  context 'On-Premises' do
```

Comment:
To avoid this context stuff, I thought on use 2 classes in the same file (what I honestly prefer 😄)

eg:

`test/unit/three_scale/analytics/user_tracking_test.rb`
```ruby
class ThreeScale::Analytics::UserTrackingOnPremisesTest < ActiveSupport::TestCase
...
end

class ThreeScale::Analytics::UserTrackingNotOnPremisesTest < ActiveSupport::TestCase
...
end
```

what do you folks prefer?

Reply:
I also prefer the 2 classes 👍 `NotOnPremises` should be `SaaS` 😂

Reply:
We could call it with SegmentDisabled?
Better than on-prem or SaaS imo

Reply:
Yeah, I wanted to explain him about the whole features thing that we have but we got busy. I will write it before the 3pm sync though 👍